src/helpers/LaneHelper.js

```diff
@@ -17,15 +17,24 @@ const LaneHelper = {
     return update(state, {lanes: {$set: updatedLanes}})
   },
 
-  appendCardsToLane: (state, {laneId, newCards, index}) => {
+  appendCardsToLane: (state, {laneId, newCards, index}, laneSortFunction) => {
     const lane = state.lanes.find(lane => lane.id === laneId)
     newCards = newCards
       .map(c => update(c, {laneId: {$set: laneId}}))
       .filter(c => lane.cards.find(card => card.id === c.id) == null)
+
     return state.lanes.map(lane => {
       if (lane.id === laneId) {
         if (index !== undefined) {
-          return update(lane, {cards: {$splice: [[index, 0, ...newCards]]}})
+          let cardsToUpdate = [...lane.cards, ...newCards]
```

Comment:
@dapi Hi!

I think your comment https://github.com/rcdexta/react-trello/pull/337#issuecomment-590458854 makes sense! However, the problem occurs because this line changes the original cards and this influences their order even when the `laneSortFunction` is not used. I suggest the code bellow:

```
if (laneSortFunction) {
  let cardsToUpdate = [...lane.cards, ...newCards]
  cardsToUpdate.sort(laneSortFunction)

  return update(lane, {cards: {$set: cardsToUpdate}})
}

return update(lane, {cards: {$splice: [[index, 0, ...newCards]]}})
```

What do you think?

Reply:
Hi @belgamo!

What is you use case to sort cards? Do you save these cards to backend? Does it sort on backend also?

Pay attention to there are event bus and we need to emit `MOVE_CARD` event to every moved card if it is resorted.

I think the better way to give possibility to customise `appendCardsToLane` function using dependency injection.

Reply:
One more thought. Can we use event MOVE_CARD to catch this moment and resort cards as we want?